packages/common/src/i18n/format_date.ts

```diff
@@ -101,6 +101,18 @@ export function formatDate(
   return text;
 }
 
+/**
+ * Return a date with full year instead of converting it to 19{yearProvided}, See:
+ * https://github.com/angular/angular/issues/40377
+ */
+function createDate(year: number, month: number, date: number): Date {
+  const newDate = new Date(year, month, date);
+  if (year >= 0 && year <= 99) {
+    newDate.setFullYear(newDate.getFullYear() - 1900);
+  }
+  return newDate;
```

Comment:
This can actually be simplified:

```suggestion
  const newDate = new Date(0);
  newDate.setFullYear(year, month, date);
  return newDate;
```

The reason that it didn't work before is that if you do not specify month and date in `setFullYear()` it retrieves the values from the current date via `getMonth()` and `getDate()`, which get messed up with locale timezone offsets etc.

Reply:
Yes exactly, done